src/app/features/fee-editor/components/custom-fee-item.tsx

```diff
@@ -5,33 +5,37 @@ import { Stack } from 'leather-styles/jsx';
 
 import { Button, Input, ItemLayout } from '@leather.io/ui';
 
-import type { FeeItemProps } from './fee-item';
+import { type EditorFee, useFeeEditorContext } from '../fee-editor.context';
+import { formatEditorFeeItem } from '../fee-editor.utils';
 import { FeeItemIcon } from './fee-item-icon';
 
-interface CustomFeeItemProps extends FeeItemProps {
-  fee: string | null;
-  setFee(fee: string): void;
+interface CustomFeeItemProps {
+  fee: EditorFee;
 }
-
-export function CustomFeeItem({
-  fee,
-  setFee,
-  feeType,
-  onSelect,
-  isSelected,
-  captionLeft,
-  titleRight,
-  captionRight,
-}: CustomFeeItemProps) {
+export function CustomFeeItem({ fee }: CustomFeeItemProps) {
   const inputRef = useRef<HTMLInputElement>(null);
+  const {
+    customEditorFeeRate,
+    marketData,
+    selectedEditorFee,
+    onSetCustomEditorFeeRate,
+    onSetSelectedEditorFee,
+  } = useFeeEditorContext();
+
+  const { captionLeft, titleRight, captionRight } = formatEditorFeeItem({
+    editorFee: fee,
+    marketData,
+  });
+
+  const isSelected = selectedEditorFee?.type === fee.type;
 
   return (
     <Button
-      onClick={() => onSelect(feeType)}
+      onClick={() => onSetSelectedEditorFee(fee)}
       variant="outline"
       borderWidth={isSelected ? '2px' : '1px'}
       borderColor={isSelected ? 'ink.border-selected' : 'ink.border-default'}
-      // Add margin compensation when not selected to maintain consistent size
+      // Add margin compensation to maintain consistent size
```

Comment:
Ideally, this would've been implemented with pseudo elements the same way `Input` was to avoid any box content overflows when applying borders.

Reply:
Yep, I can try to look at this too today.

Reply:
I'll create an issue for this since it is a change in the UI library.

Reply:
https://linear.app/leather-io/issue/LEA-2234/refactor-ui-lib-button-for-web-to-handle-border-on-selected